src/main/java/roomescape/reservation/ReservationWaitingService.java

```diff
@@ -0,0 +1,75 @@
+package roomescape.reservation;
+
+import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import roomescape.auth.dto.LoginMember;
+import roomescape.exception.custom.reason.member.MemberNotFoundException;
+import roomescape.exception.custom.reason.reservation.ReservationNotFoundException;
+import roomescape.member.Member;
+import roomescape.member.MemberRepository;
+import roomescape.reservation.reservation.Reservation;
+import roomescape.reservation.reservation.ReservationRepository;
+import roomescape.reservation.reservation.dto.ReservationAndWaitingResponse;
+import roomescape.reservation.waiting.Waiting;
+import roomescape.reservation.waiting.WaitingRepository;
+import roomescape.schedule.Schedule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class ReservationWaitingService {
+
+    private final ReservationRepository reservationRepository;
+    private final WaitingRepository waitingRepository;
+    private final MemberRepository memberRepository;
+
+    public List<ReservationAndWaitingResponse> readAllByMember(final LoginMember loginMember) {
+        Member member = getMemberByEmail(loginMember.email());
+        List<Reservation> reservations = reservationRepository.findAllByMember(member);
+        List<Waiting> waitings = waitingRepository.findAllByMember(member);
+
+        List<ReservationAndWaitingResponse> responses = new ArrayList<>();
+        reservations.stream()
+                .forEach(reservation -> responses.add(ReservationAndWaitingResponse.of(reservation)));
+        waitings.stream()
+                .forEach(waiting -> responses.add(ReservationAndWaitingResponse.of(waiting)));
```

Comment:
바로 `forEach` 사용 가능합니다.

```suggestion
        reservations.forEach(reservation -> responses.add(ReservationAndWaitingResponse.of(reservation)));
        waitings.forEach(waiting -> responses.add(ReservationAndWaitingResponse.of(waiting)));
```

Reply:
또, 빈 리스트에 `add` 해주지 않고 아래처럼 바로 할당할 수도 있습니다.
```suggestion
        List<ReservationAndWaitingResponse> responses = Stream.concat(
                reservations.stream().map(ReservationAndWaitingResponse::of),
                waitings.stream().map(ReservationAndWaitingResponse::of)
        ).toList();
```

Reply:
훨씬 깔끔하게 수정할 수 있네요! 좋은 방법 알려주셔서 감사합니다.